### PR TITLE
Add note about TemplateRenderer deprecation

### LINKED ISSUE #1190: Document migrating from TemplateRenderer to LitRenderer
`TemplateRenderer` is deprecated in favour of `LitRenderer`.
Add short migration instructions for what needs to be changed:

Before:
```java
TemplateRenderer.of("<order-card"
	+ "  header='[[item.header]]'"
	+ "  order-card='[[item.orderCard]]'"
	+ "  on-card-click='cardClick'>"
	+ "</order-card>");
```

After:
```java
LitRenderer.of("<order-card"
	+ "  .header='${item.header}'"
	+ "  .orderCard='${item.orderCard}'"
	+ "  @card-click='${cardClick}'>"
	+ "</order-card>");
```

### PR DESCRIPTION
**Pull request by tarekoraby**
Fix #1190

### CODE REVIEW

File: articles/v14-23-upgrade/final-steps/index.adoc [hunk 1]
```diff
@@ -24,6 +24,27 @@ See how the use the flag in <<{articles}/flow/configuration/properties#, Configu
...
+Would be replaced using [classname]`LitRenderer` as follows
+[source, java]
+----
+LitRenderer.of("<order-card"
+	+ "  .header='${item.header}'"
+	+ "  .orderCard='${item.orderCard}'"
+	+ "  @card-click='${cardClick}'>"
+	+ "</order-card>");
+----
+For more information, check the Grid's <<{articles}/ds/components/grid/flow#using-lit-renderers#, Lit Renderers guide>>.
```
Comment (tomivirkki) [inline articles/v14-23-upgrade/final-steps/index.adoc:46]: The link has an extra `#` at the end so it doesn't work as expected

File: articles/v14-23-upgrade/final-steps/index.adoc [hunk 2]
```diff
@@ -24,6 +24,27 @@ See how the use the flag in <<{articles}/flow/configuration/properties#, Configu
 - JavaScript execution APIs [methodname]`executeJavaScript()` and [methodname]`callFunction()` in [classname]`Element` and [classname]`Page` should be replaced with similarly named methods that give access to the return value [methodname]`executeJs()` and [methodname]`callJsFunction()`.
 - [classname]`ExecutionCanceler` is replaced with [classname]`PendingJavaScriptResult`
 - The [interfacename]`PageConfigurator` interface is deprecated; customize initial page settings instead by overriding [methodname]`AppShellConfigurator::configurePage()`.
+- [classname]`TemplateRenderer` for Grid and Combo Box (which is based on Polymer templates) is now deprecated in favor of [classname]`LitRenderer`.
+For example, the following usage of the [classname]`TemplateRenderer`
+[source, java]
```
Comment (tomivirkki) [inline articles/v14-23-upgrade/final-steps/index.adoc:29]: Syntax highlighting for this code snippet doesn't work:
Screenshot 2022-03-24 at 9 53 28

File: articles/v14-23-upgrade/final-steps/index.adoc (hunk 1 above)
Comment (tarekoraby) [reply articles/v14-23-upgrade/final-steps/index.adoc]: fixed

File: articles/v14-23-upgrade/final-steps/index.adoc (hunk 2 above)
Comment (tarekoraby) [reply articles/v14-23-upgrade/final-steps/index.adoc:31]: fixed

**Review by tomivirkki — APPROVED** [review]

**Pull request merged**